libr/anal/vtable.c

```diff
@@ -90,53 +94,56 @@ static bool vtable_section_can_contain_vtables(RVTableContext *context, RBinSect
 		r_str_endswith (section->name, "__const");
 }
 
-
-static int vtable_is_addr_vtable_start(RVTableContext *context, ut64 curAddress) {
-	RAnalRef *xref;
-	RListIter *xrefIter;
+static int vtable_is_addr_vtable_start(RVTableContext *context, ut64 curAddress, ut64 data_section_start, ut64 data_section_end) {
+	ut64 value;
 
 	if (!curAddress || curAddress == UT64_MAX) {
 		return false;
 	}
 	if (curAddress && !vtable_is_value_in_text_section (context, curAddress, NULL)) {
 		return false;
 	}
-	// total xref's to curAddress
-	RList *xrefs = r_anal_xrefs_get (context->anal, curAddress);
-	if (r_list_empty (xrefs)) {
-		r_list_free (xrefs);
+
+	if (!context->read_addr (context->anal, curAddress - context->word_size, &value)) {
+		return false;
+	}
+	if (value != 0 && (value < data_section_start || value >= data_section_end)) {
```

Comment:
I am not sure whether this holds in MSVC when RTTI is disabled. We should check that.

Reply:
yeah that's right, also for itanium this check is wrong and should allow the value to be NULL

Reply:
It does allow it, the condition is inverse.

Reply:
hah yeah sorry i need some coffee

Reply:
also !=0 is implicit, just use `value`